### ISSUE TITLE
[beta-private-cluster-update-variant] Enable image streaming at cluster level

### ISSUE THREAD

**Issue by jawnsy** [terraform-google-modules/terraform-google-kubernetes-engine#1360]
### TL;DR

The documentation describes the `enable_gcfs` flag for node pools, but this does not seem to enable the cluster-level image streaming option

### Additional information

We are defining our node pools like this, [according to the documentation](https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google/latest#node_pools-variable), with the `enable_gcfs = true` flag:

```terraform
    {
      name               = "pool"
      preemptible        = false
      spot               = true
      enable_secure_boot = true
      enable_gcfs        = true
      machine_type       = "t2d-standard-4"
      min_count          = 0
      max_count          = 10
      max_surge          = 4
      image_type         = "COS_CONTAINERD"
    },
```

However, after the cluster is created, the Image Streaming option is not enabled in the UI:

![image](https://user-images.githubusercontent.com/52710/185451006-a990b261-a840-4add-917b-cffcd27dec04.png)

**Comment by bharathkkb** [comment]
Thanks for the report! Does `enable_gcfs` show up in the plan at all? I am wondering if this is a bug in the module or the provider.

**Comment by lauraseidler** [comment]
We're having the same "issue" - but I don't think it's a bug. 

Image streaming can be enabled [on clusters](https://cloud.google.com/kubernetes-engine/docs/how-to/image-streaming#enable_on_clusters) and/or [on node pools](https://cloud.google.com/kubernetes-engine/docs/how-to/image-streaming#enable_on_node_pools), with the cluster setting being inherited to its node pools, unless the setting is overridden on node pool level.

AFAICS, this module so far only supports enabling image streaming on node pool level, so it would make sense to me that image streaming on cluster level is shown as disabled. I would expect that if the option is added to also enable it on cluster level, this would show also in the UI. I don't see image streaming in the UI for node pools, but it shows up correctly as enabled when getting it via `gcloud container node-pools describe [...]`.

**Comment by themanifold** [comment]
This shouldn't be closed, it's not fixed yet.

**Comment by snigdhasambitak** [comment]
Yes, image streaming on the cluster level is a necessary feature. So, if positive this could be an enhancement to the module

**Comment by jawnsy** [comment]
I believe this is still an open issue

**Comment by jawnsy** [comment]
I believe this is still an open issue

**Comment by jawnsy** [comment]
I think this issue is still valid, not stale.

Thanks to @lauraseidler for opening #1696!

**Comment by ProgEsteves** [comment]
@apeabody @lauraseidler
This feature seems released only for module public variants.  
Any release predictions for the module private variants as well?
